### PR TITLE
fix protobuf release steps

### PR DESCRIPTION
**Pull request by egaxhaj**
## Description

- Fixes third_party buf.build push steps to ensure that proto breaking checks don't break during releases.

closes: #1339

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by egaxhaj** [comment]
Because Provenance Protobuf files have a dependency on the third_party Protobuf library, we first need to push the third_party to BSR (https://buf.build/provenance-io) repo and update third_party/proto/buf.lock manifest with the last commit hash. This will allow for Provenance Protobuf releases to be checked agains the latest third_party release.

This workflow requires a `Personal Access Token` and a `GPG` key be set up on a code owners' GitHub account. They'll be used to sign commits used in the `create pull request` workflow step. _It is recommend that these keys be set up `in addition` to any existing keys._

1. Create a [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) (PAT) and paste it as a [secret](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets) named `CPR_PAT` (CRP stands for Create Pull Request).
3. [Generate a GPG key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-gpg-key) `RSA 4096` and export the GPG private key as an ASCII armored version to your clipboard:

```
# macOS
gpg --armor --export-secret-key joe@foo.bar | pbcopy

# Ubuntu (assuming GNU base64)
gpg --armor --export-secret-key joe@foo.bar -w0 | xclip

# Arch
gpg --armor --export-secret-key joe@foo.bar | xclip -selection clipboard -I

# FreeBSD (assuming BSD base64)
gpg --armor --export-secret-key joe@foo.bar | xclip
```
Paste your clipboard as a [secret](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets) named `GPG_PRIVATE_KEY` for example. Create another secret for the `GPG_PASSPHRASE`.

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:21]: Probably worth noting that it's a `buf.build` commit hash (not one from this repo).

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:26]: It's not helping avoid breaking changes. It's helping avoid a buf push failure due to a new third party proto reference.

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:92]: Since we should only push our protos during a release, this should really be in the `release.yaml`. That way A) we don't have to duplicate logic, and B) its in the chart when looking at the release Action in Github.

Flow:
1. The third-party protos get pushed whenever something is merged to `main`.
2. That creates a PR to update the lock.
3. That PR gets merged.
4. Things get back-ported to a release branch.
5. We tag the release.
6. Our protos get pushed, which reference the third party ones that were previously pushed and so are up-to-date.

We should take this job out of this workflow. Then, in the releases workflow, restore the `buf_push` job, but remove the step that pushes the third party protos.

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:1]: The name and filename don't need to be exact matches, but it's good to have them in sync.
```suggestion
name: Third Party Proto Registry
```
See other comments on why "Third Party" is in there.

Basically, you want it to be easy to start at Github, looking at the results of a workflow, and know which file to edit locally. Slightly less important is knowing the filename and wanting to find a run of the workflow, but it does happen.

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:3]:
```suggestion
# This workflow pushes the third party protos to buf.build whenever any of them are updated in `main`.
```

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:13]:
```suggestion
on:
  push:
    branches:
      - main
    paths:
      - "third_party/**.proto"
      - ".github/workflows/proto-registry.yml"
```

See other comment for reasoning.

**Review by SpicyLemon — CHANGES_REQUESTED** [review]
Some tweaks are needed.

File: .github/workflows/proto-registry.yml
Comment (egaxhaj) [reply .github/workflows/proto-registry.yml:92]: I'll put it back. 

I wasn't quite sure if it be better keeping the BSR push together even though they happened on different triggers (main, release).

File: .github/workflows/proto-registry.yml
Comment (egaxhaj) [reply .github/workflows/proto-registry.yml]: updated to reference `https://buf.build/provenance-io`

File: .github/workflows/proto-registry.yml
Comment (egaxhaj) [reply .github/workflows/proto-registry.yml]: updated.

File: .github/workflows/proto-registry.yml
Comment (egaxhaj) [reply .github/workflows/proto-registry.yml]: `proto_push` restored in `release.yml`

File: .github/workflows/proto-registry.yml
Comment (egaxhaj) [reply .github/workflows/proto-registry.yml]: fixed.

File: .github/workflows/proto-registry.yml
Comment (egaxhaj) [reply .github/workflows/proto-registry.yml]: done.

**Review by SpicyLemon — APPROVED** [review]

File: .github/workflows/proto-registry.yml
Comment (SpicyLemon) [inline .github/workflows/proto-registry.yml:5]: Oh! Do we want to add `workflow_dispatch` in here too: [events-that-trigger-workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)? The we have the ability to manually kick off the action if we ever need to.

I've never tried it though.

Comment (egaxhaj) [reply .github/workflows/proto-registry.yml:5]: Neither have I. The docs state that it needs to be merged to `main` (default branch) before it can be manually triggered but am not sure if it's restricted to only run on `main` after that. See step 5 in [manually-running-a-workflow](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow). As long as we can only run it on main then we're fine.

I've enabled it but let me run a test on a test repo first before we merge it.

**Review by nullpointer0x00 — APPROVED** [review]

**Review by SpicyLemon — APPROVED** [review]

**Pull request merged**
